File: evidence/generated-systems/angular/inventory-on-angular/src/app/models/TransferOrder.js

// Define collection and schema for TransferOrder
export interface TransferOrder {
    orderNumber:
	type : string
    requestedShipDate:
	type : Date
    requestedReceiveDate:
	type : Date
    shippedDate:
	type : Date
    receivedDate:
	type : Date
    OriginWarehouse:
	type : Schema.Types.ObjectId
    DestinationWarehouse:
	type : Schema.Types.ObjectId
    Lines:
 	type : [{ type: Schema.Types.ObjectId, ref: 'TransferOrderLine' }]
    Transactions:
 	type : [{ type: Schema.Types.ObjectId, ref: 'InventoryTransaction' }]
    Status:
 	type : String
#
    collection: 'transferOrders'
}

File: evidence/generated-systems/rubyonrails/inventory-on-rails/inventoryonrails/app/models/transferorder.rb
class TransferOrder < ApplicationRecord
  enum Status: [:Draft, :Released, :InTransit, :Received, :Closed, :Cancelled]


  has_many :OriginWarehouse, class_name: 'Warehouse'
  has_many :DestinationWarehouse, class_name: 'Warehouse'
  has_many :Lines, class_name: 'TransferOrderLine'
  has_many :Transactions, class_name: 'InventoryTransaction'

end
